node/src/container_chain_spawner.rs

```diff
@@ -70,11 +69,21 @@ const MAX_DB_RESTART_TIMEOUT: Duration = Duration::from_secs(60);
 /// Assuming a syncing speed of 100 blocks per second, this will take 5 minutes to sync.
 const MAX_BLOCK_DIFF_FOR_FULL_SYNC: u32 = 30_000;
 
+pub trait TSelectSyncMode:
+    Send + Sync + Clone + 'static + (Fn(bool, ParaId) -> sc_service::error::Result<SyncMode>)
```

Comment:
Using `Fn()` as a trait bound is very ugly, promise to remove this trait as soon as possible in a future PR

Reply:
Haha, as we discussed we should be able to remove `SelectSyncMode` entirely :)